File: src/YowThi.Erp.Application/Sales/SalesConfirmationOptions.cs
namespace YowThi.Erp.Application.Sales;

public sealed record SalesConfirmationOptionsQuery(string Locale, string? Search, int Offset, int Limit);
public sealed record SalesConfirmationOptionPage<T>(IReadOnlyList<T> Items, int? NextOffset);
public sealed record SalesConfirmationSaleOption(Guid Id, DateOnly SalesDate, string CustomerDisplayName, long RowVersion);
public sealed record SalesConfirmationDetailOption(Guid Id, int LineNumber, Guid SalesProductId, string ProductDisplayName, decimal Quantity, string PricingBasis, decimal? SalesWeight, decimal UnitPrice, long AmountThb);
public sealed record SalesConfirmationWorkspace(Guid SalesId, DateOnly SalesDate, Guid CustomerId, string CustomerDisplayName, long RowVersion, IReadOnlyList<SalesConfirmationDetailOption> Details);

public interface ISalesConfirmationOptionsReader
{
    ValueTask<SalesConfirmationOptionPage<SalesConfirmationSaleOption>> GetSalesAsync(SalesConfirmationOptionsQuery query, CancellationToken cancellationToken);
    ValueTask<SalesConfirmationWorkspace?> GetWorkspaceAsync(Guid salesId, string locale, CancellationToken cancellationToken);
}

File: src/YowThi.Erp.Web/src/features/sales/salesConfirmationOptions.ts
import { getApiJson } from '../../app/api/apiTransport';
import type { OperationalLocale } from '../../app/i18n/locale';

export interface SalesConfirmationSaleOption {
  id: string;
  salesDate: string;
  customerDisplayName: string;
  rowVersion: number;
}

export interface SalesConfirmationOptionsResponse {
  items: SalesConfirmationSaleOption[];
  nextCursor: string | null;
}

export interface SalesConfirmationDetail {
  id: string;
  lineNumber: number;
  salesProductId: string;
  productDisplayName: string;
  quantity: number;
  pricingBasis: string;
  salesWeight: number | null;
  unitPrice: number;
  amountThb: number;
}

export interface SalesConfirmationWorkspace {
  salesId: string;
  salesDate: string;
  customerId: string;
  customerDisplayName: string;
  rowVersion: number;
  details: SalesConfirmationDetail[];
}

interface SalesOptionsQuery {
  locale: OperationalLocale;
  search?: string;
  cursor?: string | null;
  limit?: number;
  signal?: AbortSignal;
}

export async function listSalesConfirmationOptions(
  query: SalesOptionsQuery,
): Promise<SalesConfirmationOptionsResponse> {
  const parameters = new URLSearchParams();
  const search = query.search?.trim();
  if (search) parameters.set('search', search);
  if (query.cursor) parameters.set('cursor', query.cursor);
  if (query.limit !== undefined) parameters.set('limit', String(query.limit));
  const suffix = parameters.size === 0 ? '' : `?${parameters.toString()}`;
  return getApiJson<SalesConfirmationOptionsResponse>(
    `/api/v1/sales/confirmation-options/sales${suffix}`,
    { locale: query.locale, signal: query.signal },
  );
}

export async function getSalesConfirmationWorkspace(
  salesId: string,
  locale: OperationalLocale,
  signal?: AbortSignal,
): Promise<SalesConfirmationWorkspace> {
  return getApiJson<SalesConfirmationWorkspace>(
    `/api/v1/sales/${salesId}/confirmation-workspace`,
    { locale, signal },
  );
}
